paddle/fluid/framework/details/all_reduce_op_handle.h

```diff
@@ -46,13 +49,16 @@ class AllReduceOpHandle : public OpHandleBase {
 
  protected:
   void RunImpl() override;
-
   std::vector<Scope *> local_scopes_;
+
+#if defined(PADDLE_WITH_CUDA) && !defined(_WIN32)
+#else
   std::vector<platform::Place> places_;
+#endif
+
```

Comment:
Why the `if then` is empty here?

Reply:
We can polish it in next pr to 
```
#if !defined(USE_NCCL)
#endif
```
Or change CollectiveOpHandleBase to be compatable with CPU and GPU.

Reply:
I prefer to fix it in this PR.

Reply:
Done.